.github/workflows/pypi_publish.yml

```
@@ -40,8 +53,72 @@ jobs:
+    if: startsWith(github.ref, 'refs/tags/v') || github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-latest
+    needs: [publish]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set Version (workflow_dispatch)
+        if: github.event_name == 'workflow_dispatch'
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2086:info:2:54: Double quote to prevent globbing and word splitting

(shellcheck)